### ISSUE TITLE
`update` in `TypedDict`

### ISSUE THREAD

**Issue by uf-fipper** [microsoft/pyright#7289]
## Environment data

-   Language Server version: 2024.2.2
-   OS and version: windows11
-   Python version (& distribution if applicable, e.g. Anaconda): conda 3.11.4

## Code Snippet

![image](https://github.com/microsoft/pylance-release/assets/55644019/9c67b447-c7bf-44d4-b6ac-aba9c7b4d758)

```python
from typing import TypedDict

class A(TypedDict):
    pass

a = A()
b = A()

a.update(b)
```

There is an error occurred: No overloads for "update" match the provided arguments.

But if I add a field in `A`, the error will be gone:

![image](https://github.com/microsoft/pylance-release/assets/55644019/d13b685f-c249-4ab0-b854-b0c3551de207)

Is this an expected behavior?

**Comment by erictraut** [comment]
Thanks for the bug report. This behavior was introduced when I added support for draft [PEP 705](https://peps.python.org/pep-0705/) which indicates that when all entries in a TypedDict are marked as read-only, calls to `update` should be disallowed. This is a special case, however, where there are zero entries in a TypedDict. I'm not sure why someone would define an empty TypedDict since it seems pretty useless, but it is technically legal. I've updated the logic to handle a zero-entry TypedDict as a special case.
